podman_compose.py

```diff
@@ -2070,7 +2078,42 @@ def resolve_pod_args(self) -> list[str]:
         # - Default value
         if self.global_args.pod_args is not None:
             return shlex.split(self.global_args.pod_args)
-        return self.x_podman.get("pod_args", ["--infra=false", "--share="])
+        return self.x_podman.get(
+            PodmanCompose.XPodmanSettingKey.POD_ARGS, ["--infra=false", "--share="]
+        )
+
+    def _parse_x_podman_settings(self, compose: dict[str, Any], environ: dict[str, str]) -> None:
+        known_keys = {s.value: s for s in PodmanCompose.XPodmanSettingKey}
+
+        self.x_podman = {}
+
+        for k, v in compose.get("x-podman", {}).items():
+            known_key = known_keys.get(k)
+            if known_key:
+                self.x_podman[known_key] = v
+            else:
+                log.warning(
+                    "unknown x-podman key [%s] in compose file, supported keys: %s",
+                    k,
+                    ", ".join(known_keys.keys()),
+                )
+
+        env = {
```

Comment:
duplicate implementation

Reply:
What do you mean? Duplicate of what?

Reply:
the for loop above already sets `self.x_podman[known_key] = v`. It seems that the for loop below does the same.